src/sentry/models/monitor.py

```diff
@@ -0,0 +1,138 @@
+            (cls.UNKNOWN, 'unknown'),
+            (cls.HEALTH_CHECK, 'health_check'),
+            (cls.HEARTBEAT, 'heartbeat'),
+            (cls.CRON_JOB, 'cron_job'),
+        )
+
+
+class Monitor(Model):
+    __core__ = True
+
+    guid = UUIDField(unique=True, auto_add=True)
+    organization_id = BoundedPositiveIntegerField(db_index=True)
+    project_id = BoundedPositiveIntegerField(db_index=True)
+    name = models.CharField(max_length=128)
+    status = BoundedPositiveIntegerField(
+        default=MonitorStatus.ACTIVE,
+        choices=MonitorStatus.as_choices(),
+    )
+    type = BoundedPositiveIntegerField(
+        default=MonitorType.UNKNOWN,
+        choices=MonitorType.as_choices(),
+    )
+    config = EncryptedJsonField(default=dict)
+    next_checkin = models.DateTimeField(null=True)
+    last_checkin = models.DateTimeField(null=True)
+    date_added = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        app_label = 'sentry'
+        db_table = 'sentry_monitor'
+        index_together = (('type', 'next_checkin'),)
+
+    __repr__ = sane_repr('guid', 'project_id', 'name')
+
+    def get_next_scheduled_checkin(self, last_checkin=None):
+        if last_checkin is None:
+            last_checkin = self.last_checkin
+        tz = pytz.timezone(self.config.get('timezone') or 'UTC')
+        base_datetime = last_checkin.astimezone(tz)
+        itr = croniter(self.config['schedule'], base_datetime)
+        next_checkin = itr.get_next(datetime)
+        return next_checkin + timedelta(minutes=int(self.config.get('checkin_margin') or 0))
+
+    def mark_failed(self, last_checkin=None):
+        from sentry.coreapi import ClientApiHelper
+        from sentry.event_manager import EventManager
+        from sentry.models import Project
+        from sentry.signals import monitor_failed
+
+        if last_checkin is None:
+            next_checkin_base = timezone.now()
+            last_checkin = self.last_checkin
+        else:
+            next_checkin_base = last_checkin
+
+        affected = type(self).objects.filter(
+            id=self.id,
+            last_checkin=self.last_checkin,
+        ).update(
+            next_checkin=self.get_next_scheduled_checkin(next_checkin_base),
+            status=MonitorStatus.ERROR,
+            last_checkin=last_checkin,
+        )
+        if not affected:
+            return False
+
+        event_manager = EventManager(
+            {
+                'logentry': {
+                    'message': 'Monitor failure: %s' % (self.name,),
+                },
+                'contexts': {
+                    'monitor': {
+                        'id': self.id,
+                    },
+                },
+            },
+            project=Project(id=self.project_id),
+        )
+        event_manager.normalize()
+        data = event_manager.get_data()
+        helper = ClientApiHelper(project_id=self.project_id)
+        helper.insert_data_to_database(data)
```

Comment:
also note: @alex-hofsteede @bretthoerner @tkaemming this is what i was talking about where its really hard to internally generate a real event and insert it into the pipeline. In this case I wanted to bypass rate limits and standard tracking since we wouldn't want to enforce them with this. I realize we could refactor it, but I also didn't want to be responsible for that. Something we should consider going forward within our internal APIs.